lib/private/NavigationManager.php

```
@@ -92,4 +125,61 @@ public function setActiveEntry($id) {
 	public function getActiveEntry() {
 		return $this->activeEntry;
 	}
+
+	private function init() {
+		if ($this->init) {
+			return;
+		}
+		$this->init = true;
+		if (is_null($this->appManager)) {
+			return;
+		}
+		foreach ($this->appManager->getInstalledApps() as $app) {
+			// load plugins and collections from info.xml
+			$info = $this->appManager->getAppInfo($app);
+			if (!isset($info['navigation'])) {
+				continue;
+			}
+			$nav = $info['navigation'];
+			if (!isset($nav['route'])) {
+				continue;
+			}
+			$role = isset($nav['@attributes']['role']) ? $nav['@attributes']['role'] : 'all';
+			if ($role === 'admin' && !$this->isAdmin()) {
+				continue;
+			}
+			$l = $this->l10nFac->get($app);
+			$order = isset($nav['order']) ? $nav['order'] : 100;
+			$route = $this->urlGenerator->linkToRoute($nav['route']);
+			$name = isset($nav['name']) ? $nav['name'] : ucfirst($app);
+			$icon = isset($nav['icon']) ? $nav['icon'] : 'app.svg';
+			foreach ([$icon, "$app.svg"] as $i) {
+				try {
+					$icon = $this->urlGenerator->imagePath($app, $i);
+					break;
+				} catch (\RuntimeException $ex) {
+				}
```

missing comment to justify ignoring exception